### PR TITLE
Implement Jeff's Mobile Feedback

### PR DESCRIPTION
**Pull request by Noah-Silvera**
Summary of changes that were done

- Refactored and extended the SDK section to sync up with webs, with more of a "learn the basics" approach of edit layout, creating a component, and creating a service
- Added links to relevant samples when appropiate, and fixed existing placeholder links to samples
- Updated the commands and operations article in mobile extensively to be more accurate and less jumbled in the argument passing section
- various other changes suggsted by Jeff

### CODE REVIEW

**Comment by Noah-Silvera** [comment]
Hey Jeff, some changes that didn't make it in this round were....

* Best practices section (added to the backlog)
* Disposal, memory leaks etc (backlog)
* App Startup lifecycle (backlog)
* Restructuring the order of the concepts, use cases etc (Current Sprint)

**Comment by Noah-Silvera** [comment]
@ianschmitz Could you review the changes I made to the names of the sections in the `sidebars.js` file, the restructure of the SDK section in mobile, and take a quick look at the "Relevant Samples" followed by "next steps" pattern I've adoped, such as [here](https://deploy-preview-22--geocortex-dev-center.netlify.app/docs/mobile/concepts-commands-operations#relevant-sdk-samples)

**Comment by Noah-Silvera** [comment]
@kewalker would you mind just taking a quite look at all the "Relevant Samples" headers that now exist, and see if I missed any Samples you think would be worth adding to the sections?

File: sidebars.js [hunk 1]
```diff
@@ -242,9 +228,20 @@ module.exports = {
         },
         {
             type: "category",
-            label: "Getting Started with the SDK",
+            label: "Geocortex Mobile SDK",
             items: [
-                "mobile/sdk-quick-start",
+                {
+                    type: "category",
+                    label: "Quickstart",
```
Comment (ianschmitz) [inline sidebars.js:235]: Hmm i'm not so sure anymore about the _Quickstart_ name. For consistency i'm debating on renaming that project to the _vertigis-mobile-sdk_, which also gives us consistency with Web.

Not sure what to do here - i'm not a big fan of the group label.

Comment (Noah-Silvera) [reply sidebars.js:235]: I'm not a big fan of the group label either... 
But with the upcoming work to restructure the SDK, this will likely change or altogether disappear. I think it could be left for now and tackled as part of that work. Thoughts?

File: cspell.json [hunk 2]
```diff
@@ -44,6 +45,7 @@
         "nuget",
         "polyline",
         "prereqs",
+        "prerequsites",
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply cspell.json]: Whoops. Fixed!

File: docs/mobile/concepts-commands-operations.mdx [hunk 3]
```diff
@@ -14,11 +14,7 @@ import UseCaseContainer from "../../src/components/UseCaseContainer";
...
-
-The full list of existing commands and operations available in Geocortex Mobile can be [found in the API documentation](api-commands-operations-events.mdx).
-
-Commands and Operations are grouped by namespace, for example:
+**Commands and operations** are runnable, independent units of work within Geocortex Mobile. Using the Geocortex Mobile SDK, it's possible to [define custom commands and operations](usecases-implement-command-operation.mdx) for use throughout the application. In Geocortex Mobile, commands and operations are compatible extensions of the [ICommand interface](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/commanding). Operations are implementations of `ICommand` which have been extended to have a `name` property, allowing them to be referenced globally, and an `ExecuteAsync` method, which allows the operation to return a value. Commands and Operations are grouped by namespaces, such as `auth` or `edit`.
```
Comment (jsiemensLatitudegeo) [inline docs/mobile/concepts-commands-operations.mdx:17]: It's not the Name (uppercase) property that allows them to be referenced globally. The Name property is actually only used for logging to write the name out. The IOperationRegistry (which is effectively a dictionary) allows for them to be referenced globally. The 'key' in the dictionary is the global name reference. Generally speaking, it is not necessary for the key to be in the value, nor is it necessary here. More generally, operations:

- are asynchronous
- are strongly typed
- can have return values

File: docs/mobile/sdk-advanced-quickstart.mdx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+For the most part, the Geocortex Mobile project follows the development practices outlined in [Enterprise Application Patterns using Xamarin.Forms](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/enterprise-application-patterns/).
+
+One major deviation from usual Xamarin Development patterns is that [Autofac](https://autofac.org/) is used as the Dependency Injection framework.
+
+### Geocortex Mobile SDK Packages
+
+The Geocortex Mobile SDK is composed of Geocortex packages:
+
+-   `Geocortex Icons`
```
Comment (jsiemensLatitudegeo) [inline docs/mobile/sdk-advanced-quickstart.mdx:16]: These names will need to change at some point. Maybe not part of this PR.

File: docs/mobile/concepts-advanced-app-config.mdx [hunk 5]
```diff
@@ -87,11 +113,3 @@ For example, here's references to a small, medium, and large layout in the `app.
...
 }
 ```
-
-## Custom App Items
-
-App items in Geocortex Mobile correspond to a class that extends `VisualAppItem` and registers with **Autofac** as an `AppItem` type, like the `HarmlessComponentConfiguration` in the [app config use case](usecases-implement-component-participate-app-config.mdx#create-a-skeleton-app-config-definition).
-
-<AutofacSnippet />
-
-Registering a class in this manner makes mobile aware of a new **app item type**, which can be added to the items list in the `app.json` with the appropriate `$type` specified.
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/mobile/concepts-advanced-app-config.mdx:97]: Sadly that pattern doesn't quite apply to the development methods end users have access to / we've promoted.

If you use the SDK, you **do** need to restart your app after changing the layout and config files. 

We currently don't explain how to get to the appdata folder, as that's not an ideal pattern (need to copy your changes back into the source code), and we are hoping to have some sort of live layout editing feature, as discussed in our meeting with the Cams and Michael.

If you are using the ctrl+shift+e pattern, this is the same as editing your config with the UI, and therefore refresh to reload your layout should be an established pattern you are aware of! If not, I feel like the onus should be designer to inform users about the refresh pattern.

File: docs/mobile/sdk-advanced-quickstart.mdx (hunk 4 above)
Comment (Noah-Silvera) [reply docs/mobile/sdk-advanced-quickstart.mdx:16]: Restructuring the Documentation to better organize SDK vs Configuration vs Concepts vs API content in an upcoming ticket, I'll make a note

File: docs/mobile/concepts-commands-operations.mdx (hunk 3 above)
Comment (Noah-Silvera) [reply docs/mobile/concepts-commands-operations.mdx]: Reworded to reflect this comment!
